src/Type/Unify.hs

```diff
@@ -138,10 +138,11 @@ matchArguments matchSome range free tp fixed named mbExpResTp
               then unifyError NoMatch
               else do -- trace (" matchArguments: " ++ show (map pretty pars, map pretty fixed, map pretty named)) $ return ()
                       -- subsume fixed parameters
-                      let (fpars,npars) = splitAt (length fixed) pars
+                      let parsNotNamedArg = filter (\(nm,tp) -> nm `notElem` map fst named) pars
+                      let (fpars,rest) = splitAt (length fixed) parsNotNamedArg
```

Comment:
The only other change really is during unification removing all the supplied named arguments from parameters before determining which ones are fixed.

Reply:
Other than two more one line changes just below this, all of the changes are in updating tests.